src/components/traces/searchQueryBar.jsx

```diff
@@ -91,28 +94,36 @@ export default class SearchQueryBar extends React.Component {
         event.preventDefault();
     }
 
+    isSearchValid() {
+        const queryValid = queryIsValid(this.state.queryString);
+        const dateValid = dateIsValid(this.state.startTime, this.state.endTime);
```

Comment:
date validation should be on apply inside the daterange picker component itself. 
Also, user should not be able to pick a future date, that condition can be added using config of that date picker component.